lib/config/sharding_topology_setup.js

```diff
@@ -0,0 +1,80 @@
+var f = require('util').format;
+
+// MongoDB Topology Manager
+var setupShardedCluster = function(manager) {
+  manager.addShard(
```

Comment:
`addShard` should not return a promise, and you should be able to add these sequentially, and synchronously, and just return `manager` at the bottom